### DISCUSSION TITLE
Is this model causal?

### DISCUSSION THREAD

**Discussion by andyye1999** [jhauret/eben#8] [Q&A] [answered]
when use conv1d and ConvTranspose1d

**Comment by jhauret** [comment, marked as answer]
The actual implementation is not causal but one would just have to use only past samples in convolutions to make it so.

**Reply by andyye1999** [reply]
> The actual implementation is not causal but one would just have to use only past samples in convolutions to make it so.

How did you reproduce the article "Stream SeaNet"?
and I am having issues with how to perform streaming inference.
process frame(20ms)
